vioscsi/vioscsi.c

```diff
@@ -417,16 +420,22 @@ ENTER_FN();
     adaptExt->pageOffset = 0;
     adaptExt->poolOffset = 0;
     Size = 0;
+    #define MAX_DUMP_MODE_QUEUE_NUM 256
     for (index = VIRTIO_SCSI_CONTROL_QUEUE; index < max_queues + VIRTIO_SCSI_REQUEST_QUEUE_0; ++index) {
         virtio_query_queue_allocation(&adaptExt->vdev, index, &queueLength, &Size, &HeapSize);
         if (Size == 0) {
             LogError(DeviceExtension,
                 SP_INTERNAL_ADAPTER_ERROR,
                 __LINE__);
-
-            RhelDbgPrint(TRACE_LEVEL_FATAL, ("Virtual queue %d config failed.\n", index));
+            TRACE1(TRACE_LEVEL_FATAL, DRIVER_START, "Virtual queue config failed", "queue", index);
             return SP_RETURN_ERROR;
         }
+        if (adaptExt->dump_mode && queueLength > MAX_DUMP_MODE_QUEUE_NUM) {
+           adaptExt->original_queue_num[index] = queueLength;
+           virtio_set_queue_allocation(&adaptExt->vdev, index, MAX_DUMP_MODE_QUEUE_NUM);
```

Comment:
I finally got around to looking at this in detail.

So I see that you limit the queue size to 256 when running in dump mode. What issue prompted this if I may ask? Problems allocating virtqueue memory? Why not initialize only one queue instead?

If I'm reading the code right, this:

`      max_queues = min(max_cpus, adaptExt->scsi_config.num_queues);`

means that max_queues is potentially a large number, even in dump mode.